html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/Login.pm

```diff
@@ -200,6 +201,11 @@ sub authenticate {
               }, @{$sources} );
         if (!defined $return || $return == $LOGIN_FAILURE) {
             pf::auth_log::record_auth(join(',',map { $_->id } @{$sources}), $self->current_mac, $username, $pf::auth_log::FAILED, $self->app->profile->name);
+            if ($self->actions->{'on_failure'} && @{$self->actions->{'on_failure'}} > 0) {
```

Comment:
As discussed verbally, this would be good in a function so that its easily reusable